src/ShopifyClient.php

```diff
@@ -256,24 +262,74 @@ public function __call($method, $args = [])
             ]);
         } else {
             $args = array_merge($args, [
-               '@http' => [
-                   'headers' => [
-                       'X-Shopify-Access-Token' => $this->connectionOptions['access_token']
-                   ]
-               ]
+                '@http' => [
+                    'headers' => [
+                        'X-Shopify-Access-Token' => $this->connectionOptions['access_token']
+                    ]
+                ]
             ]);
         }
 
-        $command = $this->guzzleClient->getCommand(ucfirst($method), $args);
-        $result  = $this->guzzleClient->execute($command);
+        return $this->guzzleClient->getCommand(ucfirst($method), $args);
+    }
 
-        // In Shopify, all API responses wrap the data by the resource name. For instance, using the "/shop" endpoint will wrap
-        // the data by the "shop" key. This is a bit inconvenient to use in userland. As a consequence, we always "unwrap" the result.
+    /**
+     * Execute a single command
+     *
+     * @param  CommandInterface $command
+     * @return array
+     */
+    public function execute(CommandInterface $command): array
+    {
+        $result = $this->guzzleClient->execute($command);
 
-        $operation = $this->guzzleClient->getDescription()->getOperation($command->getName());
-        $rootKey   = $operation->getData('root_key');
+        return $this->unwrapResponseData($command, $result);
+    }
+
+    /**
+     * Execute multiple commands
+     *
+     * @param  array $commands
+     * @return array
+     */
+    public function executeAll(array $commands = []): array
+    {
+        $commandResults = $this->guzzleClient->executeAll($commands);
+        $results        = [];
+
+        // Normally, results are expected to be returned in the same order as initial commands, so we can post-process them
+
+        /** @var Result $commandResult */
+        foreach ($commandResults as $index => $commandResult) {
+            // If the command has failed, we store the exception, otherwise the payload
+            $results[] = ($commandResult instanceof CommandException) ? $commandResult : $this->unwrapResponseData($commands[$index], $commandResult);
```

Comment:
Should not store in `$results[$index]`? (just in case the results comes in different order)